PyInstaller/hooks/hook-gi.repository.GdkPixbuf.py

```diff
@@ -26,116 +26,138 @@
 destpath = "lib/gdk-pixbuf-2.0/2.10.0/loaders"
 cachedest = "lib/gdk-pixbuf-2.0/2.10.0"
 
-# If the "gdk-pixbuf-query-loaders" command is not in the current ${PATH}, GDK
-# and thus GdkPixbuf is unavailable. Return with a non-fatal warning.
+# If the "gdk-pixbuf-query-loaders" command is not in the current ${PATH}, or
+# is not in the GI lib path, GDK and thus GdkPixbuf is unavailable. Return with
+# a non-fatal warning.
 gdk_pixbuf_query_loaders = None
 
-if compat.architecture == '64bit':
-    # CentOS/Fedora package as -64
-    cmds = ['gdk-pixbuf-query-loaders-64', 'gdk-pixbuf-query-loaders']
-else:
-    cmds = ['gdk-pixbuf-query-loaders']
-
-for cmd in cmds:
-    gdk_pixbuf_query_loaders = which(cmd)
-    if gdk_pixbuf_query_loaders is not None:
-        break
-
-if gdk_pixbuf_query_loaders is None:
+try:
+    libdir = get_gi_libdir('GdkPixbuf', '2.0')
+except ValueError:
     logger.warning(
-        '"hook-gi.repository.GdkPixbuf" ignored, since GDK not found '
-        '(i.e., "gdk-pixbuf-query-loaders" not in $PATH).'
+        '"hook-gi.repository.GdkPixbuf" ignored, '
+        'since GdkPixbuf library not found'
     )
-# Else, GDK is available. Let's do this.
-else:
-    binaries, datas, hiddenimports = get_gi_typelibs('GdkPixbuf', '2.0')
-    datas += collect_glib_translations('gdk-pixbuf')
-
-    libdir = get_gi_libdir('GdkPixbuf', '2.0')
 
-    # To add support for a new platform, add a new "elif" branch below with the
-    # proper is_<platform>() test and glob for finding loaders on that platform.
-    if is_win:
-        ext = "*.dll"
-    elif is_darwin or is_linux:
-        ext = "*.so"
-
-    # If loader detection is supported on this platform, bundle all detected
-    # loaders and an updated loader cache.
-    if ext:
-        loader_libs = []
-
-        # Bundle all found loaders with this user application.
-        pattern = os.path.join(libdir, loaders_path, ext)
-        for f in glob.glob(pattern):
-            binaries.append((f, destpath))
-            loader_libs.append(f)
-
-        # Sometimes the loaders are stored in a different directory from
-        # the library (msys2)
-        if not loader_libs:
-            pattern = os.path.join(libdir, '..', 'lib', loaders_path, ext)
+if libdir:
+
+    # Distributions either package gdk-pixbuf-query-loaders on the path package
+    # on the path with a special x64 version for x64 architecture or place in
+    # the GI libs directory (not on the path)
+    cmds = [
+        'gdk-pixbuf-query-loaders-64',
+        'gdk-pixbuf-query-loaders',
+        os.path.join(libdir, 'gdk-pixbuf-2.0/gdk-pixbuf-query-loaders'),
```

Comment:
Shouldn't the one in `libdir` be the preferred one? WDYT?

Reply:
I think that Debian / Ubuntu distros are the most common, so I agree that moving this up as the preferred libdir makes sense.